File: scripts/validate-env.ts
#!/usr/bin/env npx tsx
/**
 * Validate Next.js environment variables.
 * Run from project root: npx tsx scripts/validate-env.ts
 * Or: npm run validate:env
 *
 * Loads .env and .env.local, checks required vars, exits 1 if critical missing.
 */

import * as fs from "fs";
import * as path from "path";

// Load dotenv manually (script runs outside Next.js)
const root = path.resolve(__dirname, "..");
const envPath = path.join(root, ".env");
const envLocalPath = path.join(root, ".env.local");

function loadEnvFile(filePath: string): Record<string, string> {
  const env: Record<string, string> = {};
  if (!fs.existsSync(filePath)) return env;
  const content = fs.readFileSync(filePath, "utf-8");
  for (const line of content.split("\n")) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith("#")) continue;
    const eq = trimmed.indexOf("=");
    if (eq <= 0) continue;
    const key = trimmed.slice(0, eq).trim();
    let val = trimmed.slice(eq + 1).trim();
    if (val.startsWith('"') && val.endsWith('"')) val = val.slice(1, -1);
    else if (val.startsWith("'") && val.endsWith("'")) val = val.slice(1, -1);
    env[key] = val;
  }
  return env;
}

const env = { ...loadEnvFile(envPath), ...loadEnvFile(envLocalPath), ...process.env };

const REQUIRED = [
  "NEXT_PUBLIC_SUPABASE_URL",
  "NEXT_PUBLIC_SUPABASE_ANON_KEY",
  "SUPABASE_SERVICE_ROLE_KEY",
  "STRIPE_SECRET_KEY",
  "STRIPE_WEBHOOK_SECRET",
  "TELNYX_API_KEY",
] as const;

const missing: string[] = [];
for (const key of REQUIRED) {
  const val = (env[key] ?? "").trim();
  if (!val) missing.push(key);
}

if (missing.length > 0) {
  console.error("validate-env: Missing required environment variables:");
  missing.forEach((k) => console.error("  -", k));
  console.error("\nCopy deploy/env/.env.example to .env.local and fill in values.");
  process.exit(1);
}

console.log("validate-env: OK – all required Next.js env vars present");

File: scripts/validate-env.py
#!/usr/bin/env python3
"""
Validate Python backend environment variables.
Run from project root: python scripts/validate-env.py

Uses backend config (loads .env from project root), exits 1 on failure.
"""

import sys
from pathlib import Path

root = Path(__file__).resolve().parent.parent
backend = root / "backend"
sys.path.insert(0, str(backend))

from config import settings


def main() -> int:
    errors: list[str] = []

    # Voice keys (required for backend startup)
    if not (settings.deepgram_api_key or "").strip():
        errors.append("DEEPGRAM_API_KEY")
    if not (settings.grok_api_key or "").strip():
        errors.append("GROK_API_KEY")
    tts_provider = (settings.tts_provider or "elevenlabs").strip().lower()
    if tts_provider == "elevenlabs" and not (settings.elevenlabs_api_key or "").strip():
        errors.append("ELEVENLABS_API_KEY")

    # Supabase
    url = settings.get_supabase_url()
    key = (settings.supabase_service_role_key or "").strip()
    if not url or not key:
        errors.append("NEXT_PUBLIC_SUPABASE_URL (or SUPABASE_URL) and SUPABASE_SERVICE_ROLE_KEY")

    # Telnyx
    if not (settings.telnyx_api_key or "").strip():
        errors.append("TELNYX_API_KEY")

    if errors:
        print("validate-env.py: Missing required environment variables:", file=sys.stderr)
        for e in errors:
            print("  -", e, file=sys.stderr)
        print("\nCopy deploy/env/.env.example to .env.local and fill in values.", file=sys.stderr)
        return 1

    # Run config validators
    try:
        settings.validate_voice_keys()
        settings.validate_supabase()
        settings.validate_telnyx()
    except ValueError as e:
        print("validate-env.py:", e, file=sys.stderr)
        return 1

    print("validate-env.py: OK – all required backend env vars present")
    return 0


if __name__ == "__main__":
    sys.exit(main())
